core/data/src/main/java/uk/gov/gchq/gaffer/data/element/comparison/ElementEquality.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright 2018 Crown Copyright
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.gchq.gaffer.data.element.comparison;
+
+import com.google.common.collect.Sets;
+
+import uk.gov.gchq.gaffer.data.element.Edge;
+import uk.gov.gchq.gaffer.data.element.Element;
+import uk.gov.gchq.gaffer.data.element.Entity;
+import uk.gov.gchq.koryphe.tuple.predicate.KoryphePredicate2;
+
+import java.util.HashSet;
+import java.util.Objects;
+import java.util.Set;
+
+/**
+ * An {@code ElementEquality} is a {@link KoryphePredicate2} that is
+ * used to compare if two {@link Element}s are equal.  Optional {@code GroupBy} properties
+ * can be set, and will then be used within the comparison also.
+ */
+public class ElementEquality extends KoryphePredicate2<Element, Element> {
+    private Set<String> groupByProperties = new HashSet<>();
+
+
+    public ElementEquality() {
+    }
+
+    public ElementEquality(final Set<String> groupByProperties) {
+        this.groupByProperties = groupByProperties;
```

Comment:
Maybe add all these to the existing set? It's safer as then the set won't change if the caller of this constructor changes the set.